### PR TITLE
Build project

### PR DESCRIPTION
**Pull request by ShahierNashaat**
### Pull request for review create a budget app using Ruby on Rails and Rspec.

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/groups/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      <% @payments_group.each do |element|%>
+        <div class="payment">
+          <p><%= element.payment.name %></p>
+          <p>$<%= element.payment.amount %></p>
+          <%= button_to "Delete", payment_path(element.payment), :method => :delete %>
+        </div>
+      <% end %>
+    </div>
+  </div>
+
```
Comment (Dipeshtwis) [inline app/views/groups/show.html.erb]: - [ ] Please make sure to stick with the design. Include date when transaction created and also keep font small while viewing on a small screen.

### Your design
Screenshot from 2022-03-16 21-52-32

### Given design

Screenshot from 2022-03-16 21-52-14

File: spec/requests/groups_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    it 'should include correct name of group' do
+      expect(response.body).to include('shopping')
+    end
+  end
+
+  describe 'GET /new' do
+    before do
+      sign_in user
+      get new_group_path
+    end
```
Comment (Dipeshtwis) [inline spec/requests/groups_spec.rb:44]: - [ ] Some of the tests are failing. Please make sure to fix those. :slightly_smiling_face: 

Screenshot from 2022-03-16 21-59-46

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @ShahierNashaat, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights

:+1: THe project is working well.
:+1: The project is deployed on Heroku and video presentation is also good.


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/requests/groups_spec.rb (hunk 2 above)
Comment (ShahierNashaat) [reply spec/requests/groups_spec.rb:44]: They were working with but I will check them again.

File: app/views/groups/show.html.erb (hunk 1 above)
Comment (ShahierNashaat) [reply app/views/groups/show.html.erb]: Ok, thank you will do that.

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾🎉✔**
Hello @ShahierNashaat 👋
Your project is complete!  There is nothing else to say. About the linters don't worry about it as it is a general issue coming from Github. It's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**